pkg/operator/operator_test.go

```diff
@@ -0,0 +1,169 @@
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionTrue,
+			},
+		},
+		{
+			platform:          configv1.AzurePlatform,
+			expectedSyncError: true,
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionTrue,
+			},
+		},
+		{
+			platform:          bareMetalPlatform,
+			expectedSyncError: true,
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionTrue,
+			},
+		},
+		{
+			platform:          kubemarkPlatform,
+			expectedSyncError: true,
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionTrue,
+			},
+		},
+		{
+			platform: configv1.VSpherePlatform,
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionFalse,
+			},
+		},
+		{
+			platform: configv1.NonePlatform,
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionFalse,
+			},
+		},
+		{
+			platform: "bad-platform",
+			expectedConditions: map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+				configv1.OperatorAvailable:   configv1.ConditionTrue,
+				configv1.OperatorProgressing: configv1.ConditionFalse,
+				configv1.OperatorFailing:     configv1.ConditionFalse,
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(string(tc.platform), func(t *testing.T) {
+			tempDir, err := ioutil.TempDir("", "TestOperatorSync")
+			if err != nil {
+				t.Fatalf("could not create the temp dir: %v", err)
+			}
+			defer os.RemoveAll(tempDir)
+
+			images := Images{
+				MachineAPIOperator: "test-mao",
+			}
+			imagesAsJSON, err := json.Marshal(images)
+			if err != nil {
+				t.Fatalf("failed to marshal images: %v", err)
+			}
+			imagesFilePath := filepath.Join(tempDir, "test-images.json")
+			if err := ioutil.WriteFile(imagesFilePath, imagesAsJSON, 0666); err != nil {
+				t.Fatalf("could not write the images file: %v", err)
+			}
+
```

Comment:
from 108 to this line could actually go outside the loop?

Reply:
Sure, it could. Is there a compelling reason why it would be better to introduce dependency between the sub-tests by moving setup code outside of the sub-tests? The only reason that I can think of is to save the time of creating the directory and creating the images file. That does not seem like a great reason to me.

Reply:
Moved in 2dbe66c...d28b7cb.